### PR TITLE
Enhance metrics sample app with tracing

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

Create OpenTelemetry trace spans in monitoring-custom-metrics app when /metrics is called.

Changes proposed in this pull request:

- Enhance metrics sample app with tracing

**Related issue(s)**

https://github.com/kyma-project/telemetry-manager/issues/317

### CODE REVIEW

**Comment by hisarbalik** [comment]
Waiting for PR https://github.com/kyma-project/test-infra/pull/8819

**Review by hisarbalik — DISMISSED** [review]

File: .github/workflows/cutom-metrics-docker-image.yaml [hunk 1]
```diff
@@ -38,7 +38,8 @@ jobs:
       - name: Build and push Docker image
         uses: docker/build-push-action@v4
         with:
-          context: prometheus/monitoring-custom-metrics
+          context: ./prometheus/monitoring-custom-metrics
```
Comment (a-thaler) [inline .github/workflows/cutom-metrics-docker-image.yaml:41]: There is a spelling mistake in the file name

File: prometheus/monitoring-custom-metrics/custom.Dockerfile [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-ADD . /app
-WORKDIR /app
-RUN CGO_ENABLED=0 GOOS=linux go build -o main ./...
-
-FROM scratch
-LABEL source=git@github.com:kyma-project/examples.git
-COPY --from=builder /app .
-EXPOSE 8080
-CMD ["./main"]
+FROM golang:1.20.6 AS builder
```
Comment (a-thaler) [inline prometheus/monitoring-custom-metrics/custom.Dockerfile:1]: golang 1.21.0 is available, please update

File: prometheus/monitoring-custom-metrics/go.mod [hunk 3]
```diff
@@ -1,11 +1,37 @@
 module github.com/kyma-project/examples/monitoring-custom-metrics
 
-go 1.16
+go 1.20
```
Comment (a-thaler) [inline prometheus/monitoring-custom-metrics/go.mod:3]:
```suggestion
go 1.21
```

**Pull request merged**